go-controller/pkg/ovn/zone_interconnect/zone_ic_handler.go

```diff
@@ -791,6 +794,69 @@ func (zic *ZoneInterconnectHandler) getStaticRoutes(ipPrefixes []*net.IPNet, nex
 	return staticRoutes
 }
 
+// deleteStaleStaticRoutes removes static routes referencing old join or transit switch
+// IPs as either IPPrefix or NextHop for a remote zone node.
+func (zic *ZoneInterconnectHandler) deleteStaleStaticRoutes(node *corev1.Node) error {
+	var v4TransitSwitchIP, v6TransitSwitchIP, v4JoinIP, v6JoinIP *net.IPNet
+
+	nodeTransitSwitchPortIPs, err := util.ParseNodeTransitSwitchPortAddrs(node)
+	if err != nil || len(nodeTransitSwitchPortIPs) == 0 {
+		return fmt.Errorf("failed to get the node transit switch port Ips : %v", err)
+	}
+	for _, ip := range nodeTransitSwitchPortIPs {
+		if utilnet.IsIPv6CIDR(ip) {
+			v6TransitSwitchIP = ip
+		} else if utilnet.IsIPv4CIDR(ip) {
+			v4TransitSwitchIP = ip
+		} else {
+			return fmt.Errorf("invalid transit switch IP: %s", ip)
+		}
+	}
+
+	nodeGRPIPs, err := util.ParseNodeGatewayRouterJoinAddrs(node, zic.GetNetworkName())
+	if util.IsAnnotationNotSetError(err) {
+		var err1 error
+		// this is for backwards compatibility
+		nodeGRPIPs, err1 = util.ParseNodeGatewayRouterLRPAddrs(node)
```

Comment:
Are you sure we still need that?

Reply:
I am not sure. I checked other places where we are getting join IP from annotation and in all those places we are running ParseNodeGatewayRouterLRPAddrs() if IsAnnotationNotSetError() is true.